lettuce/io.py

```diff
@@ -85,26 +85,75 @@ def __call__(self, i, t, f):
                 print(err_u.item(), err_p.item(), file=self.out)
 
 
-class EnergyReporter:
-    """Reports the kinetic energy with respect to analytic solution."""
-    def __init__(self, lattice, flow, interval=1, out=sys.stdout):
+class GenericStepReporter:
+    """Generic reporter mother class - If used it reports the current iteration"""
```

Comment:
I would call this "Abstract base class for reporters that write to file or console."

Reply:
This is a nice idea.